### PR TITLE
Edge Renderer: support regular and thin instances

### PR DESCRIPTION
**Pull request by Popov72**
Also, create a single instance of the line shader for all edge renderers.

### CODE REVIEW

**Comment by Popov72** [comment]
Regarding the line shader material, maybe we should create a material for each different scene, and use the scene as a lookup to get the right material? Else, using edge renderers in different scenes won't work with the current code change...

File: src/Rendering/edgesRenderer.ts [hunk 1]
```diff
@@ -149,6 +151,32 @@ export class EdgesRenderer implements IEdgesRenderer {
     /** Gets or sets a boolean indicating if the edgesRenderer is active */
     public isEnabled = true;
 
+    private static _LineShaders: Array<ShaderMaterial> = [];
+    private static _LineShaderScenes: Array<Scene> = [];
```
Comment (sebavan) [inline src/Rendering/edgesRenderer.ts:155]: It might be simpler to keep track of it on the scene  directly ???

Comment (Popov72) [reply src/Rendering/edgesRenderer.ts]: Done.

**Comment by sebavan** [comment]
I guess we can go out of draft :-)

File: src/Rendering/edgesRenderer.ts [hunk 2]
```diff
@@ -447,7 +469,7 @@ export class EdgesRenderer implements IEdgesRenderer {
      * @return true if ready, otherwise false.
      */
     public isReady(): boolean {
-        return this._lineShader.isReady();
+        return this._lineShader.isReady(this._source, this._source.hasThinInstances);
```
Comment (deltakosh) [inline src/Rendering/edgesRenderer.ts:472]: Are we sure it is hasTHIN and not regular instances instead?

Comment (Popov72) [reply src/Rendering/edgesRenderer.ts:472]: The PR does not support regular  instances, see my comments in #6539

Comment (deltakosh) [reply src/Rendering/edgesRenderer.ts:472]: oh ok I did not read that. And how will it looks like from the user standpoint? what should I do to get instancing for my edge renderers?

Comment (Popov72) [reply src/Rendering/edgesRenderer.ts:472]: ~~You mean, "what should I do to get edge renderer for my instances"?~~

~~Nothing changed, except that a shader material is not created for each edge renderer, only a single one is created by scene.~~

Ok, I may have found a way, as I can see the current implementation for edge rendering for instances does not account for visible sub meshes, anyway.

Going to commit some changes.

Comment (Popov72) [reply src/Rendering/edgesRenderer.ts]: Committed.

I have added a new `edgesShareWithInstances` property on the mesh, to let the user tell if they want to use the edge renderer of the source mesh for all instances or not.

**Comment by sebavan** [comment]
Loooks goood !!!!

**Pull request merged**